shard_connection.cpp

```diff
@@ -542,6 +572,23 @@ void shard_connection::send_set_command(struct timeval* sent_time, const char *k
     push_req(new request(rt_set, cmd_size, sent_time, 1));
 }
 
+
+/*
+ * empty rate limit command:
+ *
+ * we send the empty rate limit command so that when a client has exceed is quota
+ * instead of writing new requests, it will just submit empty event to the event loop.
+ * This is needed since the normal rate limit based on suspending the thread that is
+ * exceeding the quota can affect the other clients on that same thread and the
+ * latency of the request as well.
+ * In this way we are not effecting other clients overall latency measurement.
+ */
+void shard_connection::send_empty_rate_limit_event() {
```

Comment:
if I understood correctly, every time that we want to suspend the requests .. we send error command(s) to the server?
isn't the latency of the command can affect the rate limit?

Reply:
@yaacovhazan-Redislabs , we write data to the file descriptor without going to the server at all. We simply write `"^\r\n"` just so that it can be discarded correctly on `void shard_connection::process_response(void)`.